lib/plotlyrb/plot_image.rb

```diff
@@ -12,14 +12,18 @@ def initialize(headers)
       @https.use_ssl = true
     end
 
-    def plot_image(data, image_path, image_type, layout = {})
-      raise "image_type #{image_type} not supported" unless VALID_IMAGE_FORMATS.include?(image_type)
-      payload = { :figure => { :data => data, :layout => layout }, :format => image_type.to_s }.to_json
+    def plot_image(plot_image_spec, image_path)
+      raise 'No :format key in spec' unless plot_image_spec.has_key?(:format)
```

Comment:
But this could do that inference 

    plot_image_spec[:format] ||= get_format_from_path(image_path)
    image_format = plot_image_spec[:format]
    raise "Image format #{image_format} not supported" unless VALID_IMAGE_FORMATS.include?(image_format.to_sym)